components/locationList.jsx

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+
+import ListItem from "@components/location/listItem";
+
+const LocationList = ({
+  closePopups,
+  locations,
+  mapObject,
+  toggleLocationList,
+}) => {
+  return (
+    <div className="list" id="location-list">
+      {locations.map((location, i) => (
+        <ListItem
+          key={`location-${location.name}-${i}`}
+          closePopups={closePopups}
+          index={i}
```

Comment:
Yeah, probably is a better way to handle large lists (like you mentioned) but not sure what it is immediately - a thing for the future :)